packages/common/src/hardhat/tasks/fxTunnel.js

```diff
@@ -0,0 +1,26 @@
+const { task, types } = require("hardhat/config");
+const MaticJs = require("@maticnetwork/maticjs")
+require("dotenv").config();
+
+// In order to receive a message on Ethereum from Polygon, `receiveMessage` must be called on the Root Tunnel contract
+// with a proof derived from the Polygon transaction hash that was checkpointed to Mainnet. 
+task("root-chain-manager-proof", "Generate proof needed to receive data from root chain manager")
```

Comment:
The logic in this task will eventually have to be encapsulated in some cron job/polling bot that should listen for `NewHeaderBlock` on the `RootChainProxy` contract deployed on Ethereum. (example deployment on Goerli [here](https://goerli.etherscan.io/address/0x2890ba17efe978480615e330ecb65333b880928e#code)). 

On each of these events, the bot should derive the proof for each transaction hash on Polygon that emitted a `MessageSent` for a contract that it is monitoring. For example, [this transaction](https://explorer-mumbai.maticvigil.com/tx/0xbb1a9589ff82229230a78ac4af0e029c5c422f3730658e3299adcf40205e1504/internal-transactions) on Mumbai emitted a `MessageSent` event and the bot should ideally be aware of this, and be ready to derive a proof for the transaction hash `0xbb1a9589ff82229230a78ac4af0e029c5c422f3730658e3299adcf40205e1504` after each `NewHeaderBlock` event is emitted from Ethereum.

Reply:
does encapsulating this logic within a task make it more difficult for us to re-use later on when dealing with this within a bot? how would you propose this logic be re-use when we get to building out this bot? would it be better to put it in a JS script that can be re-used more easily? wdyt?

Reply:
that sounds good. how do you see this task easily translating to a bot implementation? would the bot call the task? would it be a re-write into a more long-form bot-type implementation?

Reply:
I think I'd rewrite this task into a long-form bot implementation, likely deleting this task or making this task import a common script (that the bot also imports). For now I just kept this as a task for convenience.

Reply:
sgtm